public/app/plugins/datasource/influxdb/components/RawInfluxQLEditor.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, useState, useEffect } from 'react';
+import { TextArea, InlineFormLabel, Input, Select, HorizontalGroup } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { ResultFormat, InfluxQuery } from '../types';
+
+const RESULT_FORMATS: Array<SelectableValue<ResultFormat>> = [
+  { label: 'Time series', value: 'time_series' },
+  { label: 'Table', value: 'table' },
+  { label: 'Logs', value: 'logs' },
+];
+
+const DEFAULT_RESULT_FORMAT: ResultFormat = 'time_series';
+
+type Props = {
+  query: InfluxQuery;
+  onQueryChange: (query: InfluxQuery) => void;
+  onRunQuery: () => void;
+};
+
+// we handle 3 fields: "query", "alias", "resultFormat"
+// "resultFormat" changes are applied immediately
+// "query" and "alias" changes only happen on onblur
+export const RawInfluxQLEditor: FC<Props> = ({ query, onQueryChange, onRunQuery }) => {
+  const [currentQuery, setCurrentQuery] = useState(query.query);
+  const [currentAlias, setCurrentAlias] = useState(query.alias);
+
+  useEffect(() => {
+    // if the value changes from the outside, we accept it
+    if (currentQuery !== query.query) {
+      setCurrentQuery(query.query);
+    }
+  }, [query.query]);
```

Comment:
Shouldn't this trigger our hooks eslint rules? @zoltanbedi
It should work ok but I think because of the `currentQuery` the hook rules will complain. We probably use a previous value inside to manually check if the prop changed and run then.

Reply:
Nitpick: Also we could probably hide this behind a custom hook, and simplify a bit something like:
```
const [currentQuery, setCurrentQuery] = useShadowedState(query.query);
function useShadowedState (outsideVal) {
const [currentVal, setVal] = useState(outsideVal);
const prevOutsideVal = usePrev(outsideVal);
useEffect(() => {
    // if the value changes from the outside, we accept it
    if (prevOutsideVal !== outsideVal && currentVal !== outsideVal) {
      setCurrentVal(outsideVal);
    }
  }, [prevOutsideVal, outsideVal, currentVal]);
return [currentVal, setVal]

```

Reply:
> Shouldn't this trigger our hooks eslint rules?

It should and it will once https://github.com/grafana/grafana/pull/31174 is merged.